### DISCUSSION TITLE
Cannot synchronize with gitlab after the intial push

### DISCUSSION THREAD

**Discussion by ivanmilara** [GetPublii/Publii#1689] [Bug Report]
### Operating system

Windows 11 Pro

### Publii version

 0.46.1 (build 16918)

### Issue type

Application

### Bug description

When I try to synchronize with my gitlab repository (hosted in gitlab.com) I get the following error message: `"An error ocurred while connecting to the server. Response code 400 (Bad Request)" `. The remote server is not updated. 

### Steps to reproduce

1. Create a gitlab project
2. Add the .gitlab-ci.yml [recommended here](https://getpublii.com/docs/host-static-website-gitlab-pages.html)
3. Complete gitlab settings in publii
4. Synchronize the first time. Works successfully and the site is visible in Gitlab pages.
5. Do changes in Publii
6. Try to synchronize again.
7. Get the following message: 
![image](https://github.com/user-attachments/assets/24e40d67-f962-4da7-9bb7-2190b1d3718e)
8. Press "Continue and Sync"
9. Get the following error: 
![image](https://github.com/user-attachments/assets/9bf3ba8f-7ef0-43be-ac31-6c273d3c742b)
10. There is not push to gitlab. 

### Relevant log output

```shell
[Fri, 13 Sep 2024 13:21:38 GMT] (!) CLIENT CREATED
[Fri, 13 Sep 2024 13:21:40 GMT] (!) REMOTE FILE NOT DOWNLOADED
[Fri, 13 Sep 2024 13:21:40 GMT] (!) ERROR WHILE DOWNLOADING publii-files.json: undefined
Founded 59 operations to do
[Fri, 13 Sep 2024 13:21:42 GMT] (!) NO FILES TO REMOVE DETECTED
[Fri, 13 Sep 2024 13:21:42 GMT] (!) NO TEXT FILES TO UPDATE DETECTED
[Fri, 13 Sep 2024 13:21:42 GMT] (!) TEXT FILES TO UPLOAD DETECTED
[Fri, 13 Sep 2024 13:21:45 GMT] (!) COMMIT ERROR: {"name":"HTTPError","code":"ERR_NON_2XX_3XX_RESPONSE","timings":{"start":1726233703449,"socket":1726233703449,"lookup":1726233703449,"connect":1726233703449,"secureConnect":1726233703449,"upload":1726233703452,"response":1726233705215,"end":1726233705215,"phases":{"wait":0,"dns":0,"tcp":0,"tls":0,"request":3,"firstByte":1763,"download":0,"total":1766}},"description":"A file with this name already exists"}
```

**Comment by iffinland** [comment]
Im same problem

**Comment by dziudek** [comment]
@ivanmilara @iffinland - can you share a link to your gitlab repositories?

**Reply by iffinland** [reply]
Thanks for the answer, but I don't use it anymore, I found a better alternative solution.

**Comment by ivanmilara** [comment]
Hi @dziudek . Here is the link: https://gitlab.com/ivanmilara/test_fla_publi

**Comment by ivanmilara** [comment]
Any update?

**Comment by IO101010** [comment]
Exact same problem here, but with newer version of Publii. If this is the wrong place to report, please tell us.

```
deployment-process.log:
[Mon, 30 Dec 2024 14:51:42 GMT] (!) CLIENT CREATED
[Mon, 30 Dec 2024 14:51:43 GMT] (!) REMOTE FILE NOT DOWNLOADED
[Mon, 30 Dec 2024 14:51:43 GMT] (!) ERROR WHILE DOWNLOADING publii-files.json: undefined
Founded 34 operations to do
[Mon, 30 Dec 2024 14:51:55 GMT] (!) NO FILES TO REMOVE DETECTED
[Mon, 30 Dec 2024 14:51:55 GMT] (!) NO TEXT FILES TO UPDATE DETECTED
[Mon, 30 Dec 2024 14:51:55 GMT] (!) TEXT FILES TO UPLOAD DETECTED
[Mon, 30 Dec 2024 14:51:57 GMT] (!) COMMIT ERROR: {"name":"HTTPError","code":"ERR_NON_2XX_3XX_RESPONSE","timings":{"start":1735570316058,"socket":1735570316058,"lookup":1735570316059,"connect":1735570316101,"secureConnect":1735570316150,"upload":1735570316349,"response":1735570317129,"end":1735570317129,"phases":{"wait":0,"dns":1,"tcp":42,"tls":49,"request":199,"firstByte":780,"download":0,"total":1071}},"description":"A file with this name already exists"}
```

Publii version 0.46.3 (build 17008)
macOS 15.2 / M1 iMac 
GitLab repository: FreeshellO.gitlab.com/Publii-static-site

Edit: the slight difference in my case is that my initial push failed with an error 520 (reported by Publii). It turned out to be because while I had verified my GitLab by email, I hadn't yet done so by phone. Both seem to be necessary. So, I had no initial website. Follow up pushes failed and I get the exact same error messages as OP since.

**Comment by dziudek** [comment]
@ivanmilara @IO101010 - it seems that I have finally found a reason of this issue - fix will be included in v.0.46.5 planned for end of this week. I have spent few hours by analyzing upload step by step and it seems that the issue was related to creating remote files list too early. That process caused the invalid operations on files. Also I have found that after first invalid deployment, after restarting the app usually Publii was able to upload files correctly, because remote files list was correct.
